src/main/java/info/batey/kafka/unit/KafkaUnit.java

```diff
@@ -15,65 +15,116 @@
+import kafka.common.TopicAndPartition;
 import kafka.consumer.Consumer;
 import kafka.consumer.ConsumerConfig;
 import kafka.consumer.ConsumerTimeoutException;
 import kafka.consumer.KafkaStream;
 import kafka.javaapi.consumer.ConsumerConnector;
-import kafka.javaapi.producer.Producer;
 import kafka.message.MessageAndMetadata;
 import kafka.producer.KeyedMessage;
-import kafka.producer.ProducerConfig;
 import kafka.serializer.StringDecoder;
-import kafka.serializer.StringEncoder;
 import kafka.server.KafkaConfig;
 import kafka.server.KafkaServerStartable;
 import kafka.utils.VerifiableProperties;
+import kafka.utils.ZKStringSerializer$;
 import kafka.utils.ZkUtils;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.ZkConnection;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
 import org.apache.kafka.clients.producer.ProducerRecord;
-import org.apache.kafka.common.security.JaasUtils;
-
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.StringSerializer;
 import org.junit.ComparisonFailure;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import scala.collection.JavaConversions;
+import scala.collection.Seq;
+import scala.collection.mutable.Buffer;
 
 import java.io.File;
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.nio.file.Files;
-import java.util.*;
-import java.util.concurrent.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 public class KafkaUnit {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaUnit.class);
-
-    private KafkaServerStartable broker;
+    private static final Logger logger = LoggerFactory.getLogger(KafkaUnit.class);
 
+    private List<KafkaBroker> brokers;
     private Zookeeper zookeeper;
     private final String zookeeperString;
-    private final String brokerString;
+    private String brokerString;
     private int zkPort;
-    private int brokerPort;
-    private Producer<String, String> producer = null;
+    private int brokerPort = -1;
+    private KafkaProducer<String, String> producer = null;
     private Properties kafkaBrokerConfig = new Properties();
+    private int numOfBrokers;
+    private ZkClient zkClient;
+    private ZkUtils zkUtils;
+
 
-    public KafkaUnit() throws IOException {
-        this(getEphemeralPort(), getEphemeralPort());
+    public KafkaUnit() {
+        this(1);
     }
 
+    public KafkaUnit(int numOfBrokers) {
```

Comment:
Having two constructors `KafkaUnit(int)` and `KafkaUnit(int, int)` which do *very* different things is not a good idea. I'd add builder for `KafkaUnit` in such case.